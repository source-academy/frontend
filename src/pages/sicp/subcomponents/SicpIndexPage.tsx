import { H1, H2, H4 } from '@blueprintjs/core';
import React from 'react';

import SicpToc from './SicpToc';

const originalAuthors = 'Harold Abelson and Gerald Jay Sussman';
const originalWithAuthors = 'with Julie Sussman';
const adaptedAuthors = 'Martin Henz and Tobias Wrigstad';
const adaptedWithAuthors = 'with Julie Sussman';
const developers = 'Samuel Fang';

const authors = (
  <div className="sicp-authors">
    <H4>{originalAuthors}</H4>
    <p>
      {originalWithAuthors}
      <i>— original authors</i>
    </p>
    <H4>{adaptedAuthors}</H4>
    <p>
      {adaptedWithAuthors}
      <i>— adapters to JavaScript</i>
    </p>
    <H4>{developers}</H4>
    <p>
      <i>— designer and developer of this Interactive SICP JS edition</i>
    </p>
  </div>
);

const bookTitle = (
  <div>
    <H1>Structure and Interpretation of Computer Programs</H1>
    <H2>— JavaScript Edition</H2>
  </div>
);

const licenses = (
  <div className="sicp-licenses">
    <div>
      <a href="http://creativecommons.org/licenses/by-sa/4.0/" rel="nofollow">
        <img src="https://licensebuttons.net/l/by-sa/4.0/88x31.png" alt="CC BY-SA 4.0" />
      </a>
    </div>
    <div>
      <p>
        This work is licensed under a{' '}
        <a href="http://creativecommons.org/licenses/by-sa/4.0/" rel="nofollow">
          Creative Commons Attribution-ShareAlike 4.0 International License
        </a>
        .
      </p>
    </div>
    <br />
    <div>
      <a href="https://www.gnu.org/licenses/gpl-3.0.en.html" rel="nofollow">
        <img
          src="https://camo.githubusercontent.com/46d38fe6087a9b9bdf7e45458901b818765b8391/68747470733a2f2f75706c6f61642e77696b696d656469612e6f72672f77696b6970656469612f636f6d6d6f6e732f7468756d622f372f37392f4c6963656e73655f69636f6e2d67706c2e7376672f353070782d4c6963656e73655f69636f6e2d67706c2e7376672e706e67"
          alt="GPL 3"
          data-canonical-src="https://upload.wikimedia.org/wikipedia/commons/thumb/7/79/License_icon-gpl.svg/50px-License_icon-gpl.svg.png"
        />
      </a>
    </div>
    <div>
      <p>
        All JavaScript programs in this work are licensed under the{' '}
        <a href="https://www.gnu.org/licenses/gpl-3.0.en.html" rel="nofollow">
          GNU General Public License Version 3
        </a>
        .
      </p>
    </div>
    <br />
    <div>
      <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/" rel="nofollow">
        <img src="https://licensebuttons.net/l/by-nc-sa/4.0/88x31.png" alt="CC BY-NC-SA 4.0" />
      </a>
    </div>
    <div>
      <p>
        <a
          href="https://mitpress.mit.edu/books/structure-and-interpretation-computer-programs-1"
          rel="nofollow"
        >
          Print and Kindle versions of this work{' '}
        </a>
        are published by The MIT Press under a{' '}
        <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/" rel="nofollow">
          Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License
        </a>
        .
      </p>
    </div>
    <br />
  </div>
);

const SicpIndexPage: React.FC = () => {
  return (
    <div className="sicp-index-page">
      <div className="sicp-cover">
        <img src="http://source-academy.github.io/sicp/sicpjs.png" alt="SICP" />
        <div className="sicp-cover-text">
          {bookTitle}
          {authors}
        </div>
      </div>
      <br />
      <H2>Content</H2>
      <SicpToc />
      <br />
      <H2>Licenses</H2>
      {licenses}
    </div>
  );
};

export default SicpIndexPage;
