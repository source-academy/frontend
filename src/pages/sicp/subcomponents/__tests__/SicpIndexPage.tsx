import { mount } from 'enzyme';

import SicpIndexPage from '../../subcomponents/SicpIndexPage';

test('Sicp index page', () => {
  const tree = mount(<SicpIndexPage />);
  expect(tree.debug()).toMatchSnapshot();
});
