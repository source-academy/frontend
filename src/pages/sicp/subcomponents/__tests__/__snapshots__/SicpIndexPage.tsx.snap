// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sicp index page 1`] = `
<div
  className="sicp-index-page"
>
  <div
    className="sicp-cover"
  >
    <img
      alt="SICP"
      src="http://source-academy.github.io/sicp/sicpjs.png"
    />
    <div
      className="sicp-cover-text"
    >
      <div>
        <h1
          className="bp4-heading"
        >
          Structure and Interpretation of Computer Programs
        </h1>
        <h2
          className="bp4-heading"
        >
          — JavaScript Edition
        </h2>
      </div>
      <div
        className="sicp-authors"
      >
        <h4
          className="bp4-heading"
        >
          Harold Abelson and Gerald Jay Sussman
        </h4>
        <p>
          with Julie Sussman
          <i>
            — original authors
          </i>
        </p>
        <h4
          className="bp4-heading"
        >
          Martin Henz and Tobias Wrigstad
        </h4>
        <p>
          with Julie Sussman
          <i>
            — adapters to JavaScript
          </i>
        </p>
        <h4
          className="bp4-heading"
        >
          Samuel Fang
        </h4>
        <p>
          <i>
            — designer and developer of this Interactive SICP JS edition
          </i>
        </p>
      </div>
    </div>
  </div>
  <br />
  <h2
    className="bp4-heading"
  >
    Content
  </h2>
  <div
    className="sicp-toc"
  >
    <div
      className="bp4-tree sicp-toc-tree"
    >
      <ul
        className="bp4-tree-node-list bp4-tree-root"
      >
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-tree-node-caret-none"
            />
            <span
              className="bp4-tree-node-label"
            >
              Foreword
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-tree-node-caret-none"
            />
            <span
              className="bp4-tree-node-label"
            >
              Foreword to Structure and Interpretation of Computer Programs, 1984
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-tree-node-caret-none"
            />
            <span
              className="bp4-tree-node-label"
            >
              Preface
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-tree-node-caret-none"
            />
            <span
              className="bp4-tree-node-label"
            >
              Prefaces to Structure and Interpretation of Computer Programs, 1996 & 1984
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-tree-node-caret-none"
            />
            <span
              className="bp4-tree-node-label"
            >
              Acknowledgments
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-icon bp4-icon-chevron-right bp4-tree-node-caret bp4-tree-node-caret-closed"
              icon="chevron-right"
              onClick={[Function]}
            >
              <svg
                aria-labelledby="iconTitle-0"
                data-icon="chevron-right"
                height={16}
                role="img"
                viewBox="0 0 16 16"
                width={16}
              >
                <title
                  id="iconTitle-0"
                >
                  Expand group
                </title>
                <path
                  d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
            <span
              className="bp4-tree-node-label"
            >
              1 Building Abstractions with Functions
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-icon bp4-icon-chevron-right bp4-tree-node-caret bp4-tree-node-caret-closed"
              icon="chevron-right"
              onClick={[Function]}
            >
              <svg
                aria-labelledby="iconTitle-1"
                data-icon="chevron-right"
                height={16}
                role="img"
                viewBox="0 0 16 16"
                width={16}
              >
                <title
                  id="iconTitle-1"
                >
                  Expand group
                </title>
                <path
                  d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
            <span
              className="bp4-tree-node-label"
            >
              2 Building Abstractions with Data
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-icon bp4-icon-chevron-right bp4-tree-node-caret bp4-tree-node-caret-closed"
              icon="chevron-right"
              onClick={[Function]}
            >
              <svg
                aria-labelledby="iconTitle-2"
                data-icon="chevron-right"
                height={16}
                role="img"
                viewBox="0 0 16 16"
                width={16}
              >
                <title
                  id="iconTitle-2"
                >
                  Expand group
                </title>
                <path
                  d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
            <span
              className="bp4-tree-node-label"
            >
              3 Modularity, Objects, and State
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-icon bp4-icon-chevron-right bp4-tree-node-caret bp4-tree-node-caret-closed"
              icon="chevron-right"
              onClick={[Function]}
            >
              <svg
                aria-labelledby="iconTitle-3"
                data-icon="chevron-right"
                height={16}
                role="img"
                viewBox="0 0 16 16"
                width={16}
              >
                <title
                  id="iconTitle-3"
                >
                  Expand group
                </title>
                <path
                  d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
            <span
              className="bp4-tree-node-label"
            >
              4 Metalinguistic Abstraction
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-icon bp4-icon-chevron-right bp4-tree-node-caret bp4-tree-node-caret-closed"
              icon="chevron-right"
              onClick={[Function]}
            >
              <svg
                aria-labelledby="iconTitle-4"
                data-icon="chevron-right"
                height={16}
                role="img"
                viewBox="0 0 16 16"
                width={16}
              >
                <title
                  id="iconTitle-4"
                >
                  Expand group
                </title>
                <path
                  d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
            <span
              className="bp4-tree-node-label"
            >
              5 Computing with Register Machines
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-tree-node-caret-none"
            />
            <span
              className="bp4-tree-node-label"
            >
              References
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
        <li
          className="bp4-tree-node"
        >
          <div
            className="bp4-tree-node-content bp4-tree-node-content-0"
            onClick={[Function]}
            onContextMenu={[Function]}
            onDoubleClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="bp4-tree-node-caret-none"
            />
            <span
              className="bp4-tree-node-label"
            >
              About the SICP JS Project
            </span>
          </div>
          <div
            className="bp4-collapse"
            style={
              Object {
                "height": undefined,
                "overflowY": undefined,
                "transition": undefined,
              }
            }
          >
            <div
              aria-hidden={true}
              className="bp4-collapse-body"
              style={
                Object {
                  "transform": "translateY(-undefinedpx)",
                  "transition": undefined,
                }
              }
            />
          </div>
        </li>
      </ul>
    </div>
  </div>
  <br />
  <h2
    className="bp4-heading"
  >
    Licenses
  </h2>
  <div
    className="sicp-licenses"
  >
    <div>
      <a
        href="http://creativecommons.org/licenses/by-sa/4.0/"
        rel="nofollow"
      >
        <img
          alt="CC BY-SA 4.0"
          src="https://licensebuttons.net/l/by-sa/4.0/88x31.png"
        />
      </a>
    </div>
    <div>
      <p>
        This work is licensed under a
         
        <a
          href="http://creativecommons.org/licenses/by-sa/4.0/"
          rel="nofollow"
        >
          Creative Commons Attribution-ShareAlike 4.0 International License
        </a>
        .
      </p>
    </div>
    <br />
    <div>
      <a
        href="https://www.gnu.org/licenses/gpl-3.0.en.html"
        rel="nofollow"
      >
        <img
          alt="GPL 3"
          data-canonical-src="https://upload.wikimedia.org/wikipedia/commons/thumb/7/79/License_icon-gpl.svg/50px-License_icon-gpl.svg.png"
          src="https://camo.githubusercontent.com/46d38fe6087a9b9bdf7e45458901b818765b8391/68747470733a2f2f75706c6f61642e77696b696d656469612e6f72672f77696b6970656469612f636f6d6d6f6e732f7468756d622f372f37392f4c6963656e73655f69636f6e2d67706c2e7376672f353070782d4c6963656e73655f69636f6e2d67706c2e7376672e706e67"
        />
      </a>
    </div>
    <div>
      <p>
        All JavaScript programs in this work are licensed under the
         
        <a
          href="https://www.gnu.org/licenses/gpl-3.0.en.html"
          rel="nofollow"
        >
          GNU General Public License Version 3
        </a>
        .
      </p>
    </div>
    <br />
    <div>
      <a
        href="http://creativecommons.org/licenses/by-nc-sa/4.0/"
        rel="nofollow"
      >
        <img
          alt="CC BY-NC-SA 4.0"
          src="https://licensebuttons.net/l/by-nc-sa/4.0/88x31.png"
        />
      </a>
    </div>
    <div>
      <p>
        <a
          href="https://mitpress.mit.edu/books/structure-and-interpretation-computer-programs-1"
          rel="nofollow"
        >
          Print and Kindle versions of this work
           
        </a>
        are published by The MIT Press under a
         
        <a
          href="http://creativecommons.org/licenses/by-nc-sa/4.0/"
          rel="nofollow"
        >
          Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License
        </a>
        .
      </p>
    </div>
    <br />
  </div>
</div>
`;
